### PR TITLE
Introduce mastodon to social media sharing modal

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

**Comment by nlisgo** [comment]
Demo the mastodon share icon here: https://s3.amazonaws.com/ci-pattern-library/1258/patterns/04-pages-article--research-with-content-aside-with-scroll/04-pages-article--research-with-content-aside-with-scroll.html

If this approach is accessible, I need an icon which will work in these scenario's: 
- https://s3.amazonaws.com/ci-pattern-library/1258/patterns/04-pages-archive-year/04-pages-archive-year.html
- https://s3.amazonaws.com/ci-pattern-library/1258/patterns/04-pages-msa/04-pages-msa.html

Just the bordered icon and we style it differently if a content header image is set.

Also, need advice for this page: https://s3.amazonaws.com/ci-pattern-library/1258/patterns/04-pages-article--magazine/04-pages-article--magazine.html

**Comment by discodavey** [comment]
@viktoria-mahmud Just looked at this and all looks and works correctly

File: source/_patterns/00-atoms/components/social-media-sharers-journal.yaml [hunk 1]
```diff
@@ -16,12 +16,8 @@ properties:
   redditUrl:
     type: string
     format: uri
-  mastodonUrl:
-    type: string
-    format: uri
```
Comment (nlisgo) [inline source/_patterns/00-atoms/components/social-media-sharers-journal.yaml]: We need to add this back in

**Review by nlisgo — COMMENTED** [review]
We need to add the parameter mastodonUrl to https://github.com/elifesciences/pattern-library/blob/develop/source/_patterns/00-atoms/components/social-media-sharers-journal.yaml

File: source/_patterns/00-atoms/components/modal-window~share.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
... 0.149439349,17.1004442 0.022169409,12.0089343 C-0.0166226796,10.4972307 0.00729025582,9.07177343 0.00729025582,7.87958222 C0.00729025582,2.67324958 3.51199627,1.1471931 3.51199627,1.1471931 C5.27916201,0.357269448 8.31145476,0.0250851513 11.4638436,0 L11.5412949,0 Z M15.8599153,5 C14.461897,5 13.4033065,5.46697151 12.7035195,6.40103745 L12.0229667,7.39238391 L11.3425554,6.40103745 C10.642627,5.46697151 9.58403654,5 8.18615964,5 C6.97793807,5 6.00462882,5.36912752 5.26142334,6.08919046 C4.54070495,6.8092534 4.18190148,7.78253068 4.18190148,9.00730143 L4.18190148,15 L6.91358821,15 L6.91358821,9.1834452 C6.91358821,7.95732233 7.50716259,7.33498047 8.69445278,7.33498047 C10.0071899,7.33498047 10.6652556,8.0732355 10.6652556,9.5330285 L10.6652556,12.7167687 L13.3808194,12.7167687 L13.3808194,9.5330285 C13.3808194,8.0732355 14.0387436,7.33498047 15.3514807,7.33498047 C16.5387709,7.33498047 17.1323453,7.95732233 17.1323453,9.1834452 L17.1323453,15 L19.864032,15 L19.864032,9.00730143 C19.864032,7.78253068 19.5052285,6.8092534 18.7846516,6.08919046 C18.0413047,5.36912752 17.0679954,5 15.8599153,5 Z' id='Combined-Shape' fill='#212121' fill-rule='nonzero'></path></g></g></svg></a></li></ul>"
```
Comment (nlisgo) [inline source/_patterns/00-atoms/components/modal-window~share.json:4]: nitpick: add `https` before mastodon link. Annoyingly it's not valid.

File: assets/sass/patterns/atoms/social-media-sharers.scss [hunk 3]
```diff
@@ -49,11 +49,17 @@
       @include margin(0, "left");
       @include margin(6, "right");
 
+      // Temporary hardcoded
```
Comment (nlisgo) [inline assets/sass/patterns/atoms/social-media-sharers.scss:52]: I think we can remove this now?

File: assets/sass/patterns/atoms/social-media-sharers.scss [hunk 4]
```diff
@@ -51,9 +51,14 @@
...
 
-        &:first-child {
+        &:first-child,
+        &:last-child {
           display: none;
         }
+
+        &:nth-child(5) {
+          @include margin(0, "right");
+        }
```
Comment (nlisgo) [inline assets/sass/patterns/atoms/social-media-sharers.scss:61]: I thought all these changes could be reverted?

**Pull request merged**